### ISSUE TITLE
Truncated data from S3 when using AWS v1.63

### ISSUE THREAD

**Issue by nickrobinson251** [JuliaCloud/AWS.jl#474]
Here's a public dataset for which `AWSS3.s3_get_file` returns almost no data when using AWS.jl v1.63 but works as  expected with v1.62 
```sh
$ aws s3 ls "s3://invenia-public-datasets/Impute/v1.0.0/datasets.tar.gz" 
2020-10-10 20:52:15    5082788 datasets.tar.gz
```
### Using v1.63.0
```jl
(jl_u1dtGQ) pkg> st
      Status `/private/var/folders/hx/1h0bbkfd18d4n1qrnwmrl4j00000gn/T/jl_u1dtGQ/Project.toml`
  [fbe9abb3] AWS v1.63.0
  [1c724243] AWSS3 v0.9.2
  
julia> using AWS, AWSS3

julia> s3p = S3Path("s3://invenia-public-datasets/Impute/v1.0.0/datasets.tar.gz");

julia> s3_get_file(s3p.bucket, s3p.key, "163.tar.gz")

julia> stat("163.tar.gz")  # just 3 bytes
StatStruct(mode=0o100644, size=3)   
```
### Using v1.62.2
```jl
(jl_0F6HFC) pkg> st
      Status `/private/var/folders/hx/1h0bbkfd18d4n1qrnwmrl4j00000gn/T/jl_0F6HFC/Project.toml`
  [fbe9abb3] AWS v1.62.2
  [1c724243] AWSS3 v0.9.2

julia> s3p = S3Path("s3://invenia-public-datasets/Impute/v1.0.0/datasets.tar.gz");

julia> s3_get_file(s3p.bucket, s3p.key, "162.tar.gz")

julia> stat("162.tar.gz")  # expected size
StatStruct(mode=0o100644, size=5082788)
```

**Comment by nickrobinson251** [comment]
this is to do with `S3.get_object` (with the default setting) and not anything set by AWSS3 (i just happened to be using AWSS3 when i hit this) e.g.

#### v1.63.0
```jl
julia> S3.get_object(s3p.bucket, s3p.key)
3-element Vector{UInt8}:
 0x1f
 0x8b
 0x08
```

#### v1.62.2
```jl
julia> S3.get_object(s3p.bucket, s3p.key)
5082788-element Vector{UInt8}:
 0x1f
 0x8b
 0x08
    ⋮
 0x6c
 0xdd
 0x01
```

**Comment by omus** [comment]
@nickrobinson251 although this bucket is named public I don't seem to have access. Can you report back with that the 4th byte is? (e.g. `S3.get_object(s3p.bucket, s3p.key)[4]` on v1.62.2)

**Comment by nickrobinson251** [comment]
hm, sorry about that -- i will investigate!

Here are the first bytes:
```jl
julia> S3.get_object(s3p.bucket, s3p.key)[1:10]
10-element Vector{UInt8}:
 0x1f
 0x8b
 0x08
 0x00
 0xe1
 0x66
 0x7f
 0x5f
 0x00
 0x03
```

**Comment by omus** [comment]
@nickrobinson251 can you verify that the [`cv/fix-issue-474`](https://github.com/JuliaCloud/AWS.jl/tree/cv/fix-issue-474) branch fixes the issue?

**Comment by nickrobinson251** [comment]
Yes, confirmed that this fixes the issue! thanks!
```jl
julia> using AWS

julia> @service S3
Main.S3

julia> S3.get_object("invenia-public-datasets", "Impute/v1.0.0/datasets.tar.gz")
5082788-element Vector{UInt8}:
 0x1f
 0x8b
 0x08
 0x00
 0xe1
 0x66
 0x7f
 0x5f
 0x00
    ⋮
 0x1f
 0xf7
 0xb5
 0x07
 0x00
 0x6c
 0xdd
 0x01

(jl_FfM59z) pkg> st
      Status `/private/var/folders/hx/1h0bbkfd18d4n1qrnwmrl4j00000gn/T/jl_FfM59z/Project.toml`
  [fbe9abb3] AWS v1.63.1 `https://github.com/JuliaCloud/AWS.jl.git#cv/fix-issue-474`
```

**Comment by omus** [comment]
Fixed in the 1.63.1 release
